src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -103,14 +100,17 @@ object DeployPlugin extends AutoPlugin {
 
     val stageAndCacheKey = taskKey[File]("Stage the current project, and calculate its cache key.")
 
+    // Clients may override to perform a different action in addition to staging (i.e. running
+    // database migrations).
+    val stageDeploy =
```

Comment:
Now wondering about the name...

To me, anything with `stage` in the name shouldn't actually modify a remote system (i.e. migrate a database).  Could we instead have a `preDeploy` task that is empty by default?

Should this also be an input task for the general case?  Or will it get all of its information from environment variables?